src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -560,6 +568,7 @@ static Value builtinArrayConcat(ExecutionState& state, Value thisValue, size_t a
                 n += len;
                 array->setThrowsException(state, ObjectPropertyName(state.context()->staticStrings().length), Value(n), array);
             } else {
+                CHECK_ARRAY_LENGTH(n);
```

Comment:
Would you check the ES2015 [spec](https://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.concat) once again? From the spec, it says that at this point

> If n**≥**2^53-1, throw a TypeError exception.